src/app/MapManager.js

```diff
@@ -0,0 +1,79 @@
+import busStopIconUrl from '../img/bus-stop.png';
+import busIconUrl from '../img/bus.png';
+import tramIconUrl from '../img/tram.png';
+import 'leaflet/dist/leaflet.css';
+
+export class MapManager {
+
+    constructor(mapContainerId) {
+        this.mapContainerId = mapContainerId;
+      }
+
+    initializeMap(){
+        const mapContainer = document.createElement('div');
+        mapContainer.id = 'map';
+        mapContainer.style = "width: 100vw; height: 100vh";
+        document.getElementById(this.mapContainerId).append(mapContainer);
+        const L = require('leaflet');
+        const map = L.map(mapContainer,{ zoomControl: false }).setView([52.2297700, 21.0117800], 12);
+        L.control.zoom({position: 'bottomright'}).addTo(map);
+        L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+            attribution: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
+        }).addTo(map);
+        return map;
+    }
+
+    addBusStopMarker(map, busStop, listOfLines){ 
+        const busStopIcon = L.icon({iconUrl: busStopIconUrl, iconSize: [32, 32]});
+        let busStopMarker;
+        if(busStop.lat != "null" && busStop.lng != "null"){
+            map.flyTo([busStop.lat,busStop.lng], 17);
+            busStopMarker = new L.marker([busStop.lat,busStop.lng], {icon: busStopIcon})
+            .addTo(map)
+            .bindPopup(`${busStop.name} ${busStop.number} <br> Lines: ${listOfLines}`);
+        }
+        return busStopMarker;
+    };
+
+    setVehicleMarkers(map, vehicleType, vehicles, vehicleMarkers){
+        let vehicleIcon;
+        if(vehicleType ==='bus'){
+           vehicleIcon = L.icon({iconUrl: busIconUrl, iconSize: [24, 24]});
+        } else if(vehicleType ==='tram'){
+            vehicleIcon = L.icon({iconUrl: tramIconUrl, iconSize: [24, 24]});
+         } else{
+             throw new Error("Unable to track this vehicle.");
+         }
+
+        let zoomBoundsToBeSet;
+        if(vehicleMarkers.length === 0 && vehicles.length > 0){
+            zoomBoundsToBeSet = true;
+        }
```

Comment:
```suggestion
        let zoomBoundsToBeSet = vehicleMarkers.length === 0 && vehicles.length > 0;
```
Much better equivalent (the best thing would be to encapsulate this logic to separate function with proper name)